charset/decode/decode.ts

```diff
@@ -0,0 +1,15 @@
+namespace $ {
+	const TextDecoder = globalThis.TextDecoder ?? $node.util.TextDecoder
+
+	export type $mol_charset_encoding = 'utf8' | 'ibm866' | 'iso-8859-2' | 'iso-8859-3' | 'iso-8859-4' | 'iso-8859-5' | 'iso-8859-6'
+		| 'iso-8859-7' | 'iso-8859-8' | 'iso-8859-8i' | 'iso-8859-10' | 'iso-8859-13' | 'iso-8859-14' | 'iso-8859-15' | 'iso-8859-16'
+		| 'koi8-r' | 'koi8-u' | 'koi8-r' | 'macintosh' | 'windows-874' | 'windows-1250' | 'windows-1251' | 'windows-1252' | 'windows-1253'
+		| 'windows-1254'| 'windows-1255'| 'windows-1256'| 'windows-1257'| 'windows-1258' | 'x-mac-cyrillic' | 'gbk'
+		| 'gb18030' | 'hz-gb-2312' | 'big5' | 'euc-jp' | 'iso-2022-jp' | 'shift-jis' | 'euc-kr' | 'iso-2022-kr' | 'utf-16be' | 'utf-16le'
+		| 'x-user-defined' | 'replacement'
```

Comment:
А что это за 2 странных кодировки?

Reply:
The legacy miscelleanous encodings: 'utf-16be', 'utf-16le', and 'x-user-defined'.
A special encoding, 'replacement', which only emits an error and an EOF code point. It is used to prevent attacks that mismatch encodings between the client and server. It can happen with ISO-2022-CN and ISO-2022-CN-ext.

https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder/encoding

Reply:
Ничего не понял, но кажется легаси нам не нужен.

Reply:
удалил